tests/providers/kubernetes/test_base.py

```diff
@@ -0,0 +1,148 @@
+    self.assertTrue(mock_k8s_api_func.called)
+    # Assert returned nodes correspond to provided response
+    self.assertEqual(set(node.name for node in nodes),
+                     set(node.metadata.name for node in mock_nodes.items))
+
+  @typing.no_type_check
+  @mock.patch('kubernetes.client.CoreV1Api')
+  def testClusterListPods(self, mock_k8s_api):
+    """Test that pods of a cluster are correctly listed."""
+
+    # Create and assign mocks
+    mock_pods = k8s_mocks.MakeMockNodes(5)
+    mock_k8s_api_func = mock_k8s_api.return_value.list_pod_for_all_namespaces
+    mock_k8s_api_func.return_value = mock_pods
+
+    pods = base.K8sCluster(api_client=k8s_mocks.MOCK_API_CLIENT).ListPods()
+
+    # Assert API and corresponding function was called appropriately
+    self.assertTrue(mock_k8s_api.called_with(k8s_mocks.MOCK_API_CLIENT))
+    self.assertTrue(mock_k8s_api_func.called)
+    # Assert returned pods correspond to provided response
+    self.assertEqual(set(pod.name for pod in pods),
+                     set(pod.metadata.name for pod in mock_pods.items))
+
+  @typing.no_type_check
+  @mock.patch('kubernetes.client.CoreV1Api')
+  def testClusterListNamespacedPods(self, mock_k8s_api):
+    """Test that namespaced pods of a cluster are correctly listed."""
+
+    # Create and assign mocks
+    mock_namespace = mock.Mock()
+    mock_pods = k8s_mocks.MakeMockNodes(5)
+    mock_k8s_api_func = mock_k8s_api.return_value.list_namespaced_pod
+    mock_k8s_api_func.return_value = mock_pods
+
+    pods = base.K8sCluster(api_client=k8s_mocks.MOCK_API_CLIENT).ListPods(
+      mock_namespace
+    )
+
+    # Assert API and corresponding function was called appropriately
+    self.assertTrue(mock_k8s_api.called_with(k8s_mocks.MOCK_API_CLIENT))
+    self.assertTrue(mock_k8s_api_func.called_with(mock_namespace))
+    # Assert returned pods correspond to provided response
+    self.assertTrue(set(pod.name for pod in pods),
+                    set(pod.metadata.name for pod in mock_pods.items))
+
+class K8sNodeTest(unittest.TestCase):
+  """Test K8sCluster functionality, mainly checking API calls."""
+
+  @typing.no_type_check
+  @mock.patch('kubernetes.client.CoreV1Api')
+  def testNodeListPods(self, mock_k8s_api):
+    """Test that pods on a node are correctly listed."""
+
+    # Create and assign mocks
+    mock_node_name = 'fake-node-name'
+    mock_pods = k8s_mocks.MakeMockNodes(5)
+    mock_k8s_api_func = mock_k8s_api.return_value.list_pod_for_all_namespaces
+    mock_k8s_api_func.return_value = mock_pods
+
+    pods = base.K8sNode(k8s_mocks.MOCK_API_CLIENT, mock_node_name).ListPods()
+
+    # Assert API and corresponding function was called appropriately
+    self.assertTrue(mock_k8s_api.called_with(k8s_mocks.MOCK_API_CLIENT))
+    self.assertTrue(mock_k8s_api_func.called)
+    kwargs = mock_k8s_api_func.call_args.kwargs
+    self.assertIn('field_selector', kwargs)
+    self.assertIn(mock_node_name, kwargs['field_selector'])
```

Comment:
Here you should favour writing your assertion with the full string you are expecting, this makes it easier to read the test:
`self.assertIn('fake-node-name', kwargs['field_selector'])`